libavcodec/qsvdec.c

```diff
@@ -616,6 +616,8 @@ int ff_qsv_process_data(AVCodecContext *avctx, QSVContext *q,
     }
 
     if (!q->initialized) {
+        if (ret < 0)
```

Comment:
is it possible to move  this to line 616?

Reply:
I think no. If this is moved out of this conditional statement.  It will skip all the non-key frames all the time. It only need to skip frames when it reinitialize decoder.

Reply:
better add some comment here, explain why we not return before this

Reply:
Ok